torchao/quantization/quantize_/workflows/int4/int4_xpu_tensor.py

```diff
@@ -0,0 +1,182 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD 3-Clause license found in the
+# LICENSE file in the root directory of this source tree.
+
+
+from typing import List
+
+import torch
+
+from torchao.quantization.quant_primitives import (
+    MappingType,
+    _choose_qparams_affine,
+    _quantize_affine,
+)
+from torchao.utils import (
+    TorchAOBaseTensor,
+)
+
+__all__ = [
+    "Int4XPUTensorIntZP",
+]
+
+aten = torch.ops.aten
+
+
+class Int4XPUTensorIntZP(TorchAOBaseTensor):
+    """
+    int4 weight-only quantization on XPU with oneDNN as backend (groupwise quantization only)
+
+    Tensor Attributes:
+        qdata: packed int4 weigh, always viewed as a 2D (N, K/2) tensor, last dimension is packed
+               preshuffling is specific to CPU kernels, see Note below.
```

Comment:
is this correct? can you update this to describe how xpu tensor is packed? examples can be found in https://github.com/pytorch/ao/blob/c93bc7d2725e0f5af1c266d90f6f0838c17c4e92/torchao/quantization/quantize_/common/packing_format.py#L18